ext/node/ops/ipc.rs

```diff
@@ -134,106 +372,161 @@ mod impl_ {
   }
 
   impl IpcJsonStreamResource {
-    pub fn new(stream: i64) -> Result<Self, std::io::Error> {
+    pub fn new(
+      stream: i64,
+      ref_tracker: IpcRefTracker,
+    ) -> Result<Self, std::io::Error> {
       let (read_half, write_half) = pipe(stream as _)?;
       Ok(Self {
         read_half: AsyncRefCell::new(IpcJsonStream::new(read_half)),
         write_half: AsyncRefCell::new(write_half),
         cancel: Default::default(),
+        queued_bytes: Default::default(),
+        ref_tracker,
       })
     }
 
-    #[cfg(unix)]
-    #[cfg(test)]
-    fn from_stream(stream: UnixStream) -> Self {
+    #[cfg(all(unix, test))]
+    fn from_stream(stream: UnixStream, ref_tracker: IpcRefTracker) -> Self {
       let (read_half, write_half) = stream.into_split();
       Self {
         read_half: AsyncRefCell::new(IpcJsonStream::new(read_half)),
         write_half: AsyncRefCell::new(write_half),
         cancel: Default::default(),
+        queued_bytes: Default::default(),
+        ref_tracker,
       }
     }
 
-    #[cfg(windows)]
-    #[cfg(test)]
-    fn from_stream(pipe: NamedPipeClient) -> Self {
+    #[cfg(all(windows, test))]
+    fn from_stream(
+      pipe: NamedPipeClient,
+      ref_tracker: ExternalOpsTracker,
+    ) -> Self {
       let (read_half, write_half) = tokio::io::split(pipe);
       Self {
         read_half: AsyncRefCell::new(IpcJsonStream::new(read_half)),
         write_half: AsyncRefCell::new(write_half),
         cancel: Default::default(),
+        queued_bytes: Default::default(),
+        refed: AtomicBool::new(false),
+        ref_tracker,
       }
     }
 
-    async fn write_msg(
+    /// writes _newline terminated_ JSON message to the IPC pipe.
+    async fn write_msg_bytes(
       self: Rc<Self>,
-      msg: serde_json::Value,
+      msg: &[u8],
+    ) -> Result<(), AnyError> {
+      let mut write_half =
+        RcRef::map(self, |r| &r.write_half).borrow_mut().await;
+      write_half.write_all(msg).await?;
+      Ok(())
+    }
+
+    #[cfg_attr(not(test), allow(dead_code))]
```

Comment:
Is this attr still needed?

Reply:
Yeah the tests use this function but the impl doesn't.

Reply:
Only a few usages, just updated them and removed this